### PR TITLE
docs: run: update --no-exec description

### PR DESCRIPTION
**Pull request by efiop**
Per https://github.com/iterative/dvc/pull/3729

> You may disregard these recommendations if you used the **Edit on GitHub** button from dvc.org to improve a doc in place.

❗ Please read the guidelines in the [Contributing to the Documentation](https://dvc.org/doc/user-guide/contributing/docs) list if you make any substantial changes to the documentation or JS engine.

🐛 Please make sure to mention `Fix #issue` (if applicable) in the description of the PR. This causes GitHub to close it automatically when the PR is merged.

Please chose to [allow us](https://help.github.com/en/github/collaborating-with-issues-and-pull-requests/allowing-changes-to-a-pull-request-branch-created-from-a-fork) to edit your branch when creating the PR.

Thank you for the contribution - we'll try to review it as soon as possible. 🙏

### CODE REVIEW

File: content/docs/command-reference/run.md [hunk 1]
```diff
@@ -153,8 +153,9 @@ data pipeline (e.g. random numbers, time functions, hardware dependency, etc.)
 - `--no-exec` - create a stage file, but do not execute the `command` defined in
   it, nor track dependencies or outputs with DVC. In the DVC-file contents, the
   file hash values will be empty; They will be populated the next time this
-  stage is actually executed. This is useful if, for example, you need to build
-  a pipeline (dependency graph) first, and then run it all at once.
+  stage is actually executed. DVC will add your outputs to `.gitignore` to
+  prevent possible future conflicts. This is useful if, for example, you need to
+  build a pipeline (dependency graph) first, and then run it all at once.
```
Comment (efiop) [inline content/docs/command-reference/run.md:158]: Clearly could be worded/explained better. Suggestions are welcome :slightly_smiling_face:

Comment (shcheklein) [reply content/docs/command-reference/run.md:158]: not sure I understand this `to prevent possible future conflicts` to even suggest improvements :)

**Comment by efiop** [comment]
Updated the message.

It might be considered an implementation detail even, since it is the same thing that it would do without `--no-exec`. But still seems to be worth pointing out.

**Comment by shcheklein** [comment]
Thanks @efiop !

**Pull request merged**